### PR TITLE
Fix import errors when Don't override value is enabled

### LINKED ISSUE #11080: Import errors
### Mautic Version

4.2.x series

### PHP version

7.4.3

### What browsers are you seeing the problem on?

Not relevant

### What happened?

`[2022-04-17 17:47:04] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "strpos() expects parameter 1 to be string, int given" at /var/www/html/app/bundles/LeadBundle/EventListener/ImportContactSubscriber.php line 198 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): strpos() expects parameter 1 to be string, int given at /var/www/html/app/bundles/LeadBundle/EventListener/ImportContactSubscriber.php:198)"} []`

Receiving import errors that weren't there prior to this PR:  https://github.com/mautic/mautic/pull/10884

### How can we reproduce this issue?

Create a simple .csv file with a single header line (firstname, lastname, email) and a single contact to import (john, smith, test@test.com) import from csv and select import in background.  I've been able to (after feeding mautic the same file multiple times) get the file to successfully import with no changes so this seems like a race condition.

### Relevant log output

```shell
`[2022-04-17 17:47:04] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "strpos() expects parameter 1 to be string, int given" at /var/www/html/app/bundles/LeadBundle/EventListener/ImportContactSubscriber.php line 198 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): strpos() expects parameter 1 to be string, int given at /var/www/html/app/bundles/LeadBundle/EventListener/ImportContactSubscriber.php:198)"} []`
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://app.bountysource.com/issues/108497370-import-errors?utm_campaign=plugin&utm_content=tracker%2F5355074&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://app.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F5355074&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by biozshock**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ 4.3 ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ x ] 
| Issue(s) addressed                     | Fixes #11080 

#### Description:

Check for variable type before passing it into `strpos`

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a simple CSV `john, smith, test@test.com`
3. Import this as a contact in background.

You may refer to the video in https://github.com/mautic/mautic/issues/11080#issuecomment-1129209590

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11350"></a>

### CODE REVIEW

**Review by dlackovic2 — DISMISSED** [review]
I've tested the fix in Gitpod by importing a dummy .csv which would fail on current version and I can confirm that the issue is fixed.

**Comment by RCheesley** [comment]
Awesome, thanks for reviewing the PR @dlackovic2 🚀

**Comment by RCheesley** [comment]
@all-contributors please add @biozshock for code

**Comment by RCheesley** [comment]
@all-contributors please add @dlackovic2 for userTesting

**Review by escopecz — DISMISSED** [review]
This is good to go. Thanks for the amazing test! 👍

File: app/bundles/LeadBundle/Tests/EventListener/ImportContactSubscriberTest.php [hunk 1]
```diff
@@ -61,6 +61,42 @@ public function getFieldList(bool $byGroup = true, bool $alphabetical = true, ar
...
+     * @see https://github.com/mautic/mautic/issues/11080
+     */
+    public function testHandleFieldWithIntValues(): void
+    {
+        $formMock = $this->createMock(Form::class);
+        $formMock->method('getData')
+            ->willReturn(
+                [
+                    'name' => 'Bud',
+                    'skip_if_exists' => 1,
```
Comment (escopecz) [inline app/bundles/LeadBundle/Tests/EventListener/ImportContactSubscriberTest.php]: SC Fixer fix:
```suggestion
                    'name'           => 'Bud',
                    'skip_if_exists' => 1,
```

**Review by dlackovic2 — APPROVED** [review]

**Review by escopecz — APPROVED** [review]

**Comment by escopecz** [comment]
@biozshock @dlackovic2 I'm blown away how good work you are doing and the reaction times are unbelievable! It was pleasure working with you both. I hope it's not the last time.

**Comment by escopecz** [comment]
@all-contributors please add @biozshock for code

**Comment by escopecz** [comment]
https://github.com/all-contributors please add @dlackovic2 for tests

**Comment by escopecz** [comment]
@all-contributors please add @dlackovic2 for tests

**Pull request merged**